plugin-cgroupv2/src/k8s/utils.rs

```diff
@@ -468,37 +622,494 @@ mod tests {
+            Some("/tmp/test-alumet-plugin-k8s/var_6/run/secrets/kubernetes.io/serviceaccount/token_6".to_string());
+
+        let result = get_pod_name(uid, node, "", &token).await;
+        assert!(result.is_ok());
+
+        let (name, namespace, node) = result.unwrap();
+        assert!(name.is_empty());
+        assert!(namespace.is_empty());
+        assert!(node.is_empty());
+
+        std::fs::remove_dir_all(&root).unwrap();
+    }
+
+    // Test `get_pod_name` with JSON value send in fake server to a specific token
+    #[tokio::test]
+    async fn test_get_pod_name_with_server_request() {
+        let tmp = std::env::temp_dir();
+        let root = tmp.join("test-alumet-plugin-k8s/var_7/");
+
+        if root.exists() {
+            std::fs::remove_dir_all(&root).unwrap();
+        }
+
+        let dir = root.join("run/secrets/kubernetes.io/serviceaccount/");
+        std::fs::create_dir_all(&dir).unwrap();
+
+        let content = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwiZXhwIjo0MTAyNDQ0ODAwLCJuYW1lIjoiVDNzdDFuZyBUMGszbiJ9.3vho4u0hx9QobMNbpDPvorWhTHsK9nSg2pZAGKxeVxA";
+        let path = dir.join("token_7");
+        std::fs::write(&path, content).unwrap();
+
+        let node = "pod1";
+        let url = format!("/api/v1/pods/?fieldSelector=spec.nodeName={}", node);
+        let _mock = mock("GET", url.as_str())
+            .with_status(200)
+            .with_header("Content-Type", "application/json")
+            .with_body(
+                json!({
+                    "items": [
+                        {
+                            "metadata": {
+                                "name": "pod1",
+                                "namespace": "default",
+                                "uid": "01234",
+                                "annotations": {
+                                    "kubernetes.io/config.hash": "hash1"
+                                }
+                            },
+                            "spec": {
+                                "nodeName": "node1"
+                            }
+                        },
+                        {
+                            "metadata": {
+                                "name": "pod2",
+                                "namespace": "default",
+                                "uid": "56789"
+                            },
+                            "spec": {
+                                "nodeName": "node2"
+                            }
+                        }
+                    ]
+                })
+                .to_string(),
+            )
+            .create();
+
+        let uid = "hash1";
+        let kubernetes_api_url = &mockito::server_url();
+        let mut token = Token::new(TokenRetrieval::File);
+
+        token.path =
+            Some("/tmp/test-alumet-plugin-k8s/var_7/run/secrets/kubernetes.io/serviceaccount/token_7".to_string());
+
+        get_pod_name(uid, node, kubernetes_api_url, &token).await.unwrap();
```

Comment:
The result of the function should be checked, otherwise the test does not prevent regressions properly.